packages/ui/src/hooks/useApi.js

```diff
@@ -15,13 +15,15 @@ export const useApi = (config) => {
   const [isSuccess, setIsSuccess] = useState(false);
   const [loading, setLoading] = useState(false);
 
-  const makeRequest = async () => {
+  const makeRequest = async (dynamicConfig = {}) => {
```

Comment:
Have you evaluated your scenario and compared other components that justifies the need for this?